### DISCUSSION TITLE
Saving data for free-sort plugin

### DISCUSSION THREAD

**Discussion by jaydavies16129** [jspsych/jsPsych#2007] [Q&A] [answered]
Hi everyone,

I'm having some trouble saving one aspect of the free-sort plugin, the prompt, to the data output. 

I dont know if this is something I need to amend in the plugin, e.g.:

```
const trial_data = {
          init_locations: init_locations,
          moves: moves,
          final_locations: final_locations,
          rt: rt
        };
```
or within the trial code?

I am running the study on Gorilla, and have the following output: 
![image](https://user-images.githubusercontent.com/84909379/126489820-7e77fc7e-caf1-4abe-b3e4-8fb78a18a2f1.jpeg)

Here is an example trial so you can see what the prompt should be in the output:
```
var trial_1 = {
        type: 'free-sort',
        stimuli: images,
        stim_height: 140, 
        stim_width: 165,
        scale_factor: 1,
        sort_area_shape: "square",
        sort_area_height: 550,
        sort_area_width: 970,
        change_border_background_color: false,
        border_width: 5,
        prompt: '<p style="font-size:25px" ><b>Places like this are fascinating.   </b></p>',
        prompt_location: "above",
        column_spread_factor: -1.5,
        stim_starts_inside: false,
        on_load: function() {
          // after the trial loads, get the sort area element on the page
          var sort_element = document.getElementById('jspsych-free-sort-arena');
          // add label above it
          var label_above = '<p style="color: green;font-size: 20px";"><b>Strongly Agree</b></p>'
          sort_element.insertAdjacentHTML('beforebegin',label_above);
          // add label below it
          var label_below = '<p style="color: red;font-size: 20px";"><b>Strongly Disagree</b></p>';
          sort_element.insertAdjacentHTML('afterend',label_below);
        }
      };
  ```
Thanks in advance!!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @jaydavies16129, you're right that the plugin doesn't save the prompt by default. So one way to change this is to modify the plugin file. You're right that this is the relevant bit of plugin code: https://github.com/jspsych/jsPsych/blob/db8bab4c369ce930b15e1bd5e6bd9ed3369fc006/plugins/jspsych-free-sort.js#L398-L403

You can add the prompt to the data by adding a new 'prompt' property to the trial data object and giving it the value `trial.prompt`:
```js
const trial_data = {
  init_locations: init_locations,
  moves: moves,
  final_locations: final_locations,
  rt: rt,
  prompt: trial.prompt
};
```

Or if you're using jsPsych 6.3.0+, then you can use the [`save_trial_parameters` option](https://www.jspsych.org/overview/plugins/#the-save_trial_parameters-parameter) that is now available in all trials. The benefit of doing it this way is that you don't need to change the plugin file. Here's what your trial would look like:
```js
var trial_1 = {
  type: 'free-sort',
  // other parameters ...
  save_trial_parameters: {
    prompt: true
  }
};
```
I hope that helps!

**Comment by jaydavies16129** [comment]
hi @becky-gilbert, you're a star as always!! 
Sometimes it really is as simple as one line. I had tried the trial.prompt but in the wrong place/paired with something else.
Thanks so much, this now saves the prompt!

**Reply by becky-gilbert** [reply]
No problem, we've all been there! 😄
